tutorial/tutorial-template-guide.md

```diff
@@ -0,0 +1,157 @@
+
+### About the "Before you begin" section
+
+Use this section to list any prerequisites for completing the tutorial. This may include operating systems, languages, package managers, or software.
+
+### About the "Steps" section
+
+The steps section is where you describe what the user needs to do to complete the tutorial. The template organizes steps in the following way:
+
+```
+1. {Write the command here. Start with a verb.}
+
+   {Optional: Explanatory text}
+
+   {Optional: Code sample or screenshot that helps your users complete this step}
+
+   {Optional: Result}
+
+```
+
+The first element is the step itself, like "Enter a name for your new repository." When writing steps, keep these tips in mind:
+
+* Start with an imperative verb. For example, "connect", "set up", or "arrange." Don't use the *-ing* form of the verb because it is harder to translate.
+* Express steps as a complete thought. Instead of saying, "Set up access", you might say, "Set up access to a Cloud Storage bucket."
+
+Next, you may choose to add some additional information about the step. Continuing with the example, you might provide some best practices for creating memorable repository names.
+
+Optionally, you can add a code sample or screenshot after the explanatory text, depending on the type of tutorial you're writing. Screenshots are a great way to show specific parts of the screen you are referring to in a step. 
+
+If you're including code samples in your repository, make sure to add comments to your code to help learners understand what each part of the code does. Additionally, make sure your code samples work, as learners will copy and paste your code to use in their own projects.
+
+The last, and optional, element in a step, is the result. Continuing with the example, you might describe what happens after a user clicks the "Create repository" button.
+
+Here is an example step:
+
+```
+1. Enter a name for your new repository.   
+
+   Good repository names are short and self-explanatory. Avoid repository names with three or more words.
+   
+   After you click "Create repository", GitHub creates your repository and the main page for the repository is displayed.
+```
+
+If needed, you can add substeps below a primary step. Make sure to indent the substep one tab space over if you're using Markdown:
+
+```
+1. Create a new pull request.
+
+   a. Navigate to the main page of your repository. 
+
+   b. Under your repository name, click **Pull requests**.
+
+      By default, all open pull requests are displayed.
+
+   c. ....
+```
+
+If you're including code samples in your steps, make sure they are also indented correctly:
+
+1. Set your Git username for your repository. 
+   
+   You can change the name that is associated with your Git commits using the `git config` command.
+
+   ```bash
+   git config user.name "Dakota Everson"
+   ```
+
```

Comment:
Suggestion:
I think it might be worth adding some notes related to maintenance. It might even be worth it's own heading.
```suggestion

Note: Consider whether your team will have the bandwidth for the long term maintenance of this content.
* A short tutorial which is up-to-date is usually better than a more detailed but unmaintained and broken tutorial.
* If you add a code sample, how will it be tested with each new release to ensure the code sample still works?
* Will screen shots be checked and updated with each software release?
* Reducing the size of the screenshot to just the area of interest can reduce the maintenance load.
* Screenshots of just the application, without the background window, might make your documentation agnostic of the underlying operating system.
* Is your documentation being translated? If so, consider how screenshots will be maintained in multiple languages.
* Adding annotations to screenshots can make them easier to understand, but it adds significantly to the ongoing maintenance overhead.

```